src/NuGet.Core/NuGet.Commands/RestoreCommand/IProject.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace NuGet.Commands.Restore
+{
+    /// <summary>
+    /// Restore has many entry points, and each gets project items and properties with different APIs.
+    /// This interface abstracts the project model to allow a single implementation of
+    /// <see cref="MSBuildRestoreUtility.GetPackageSpec(IProject)"/> to be shared across all restore entry points.
+    /// </summary>
+    public interface IProject
+    {
+        /// <summary>The full, absolute path to the project file.</summary>
+        string FullPath { get; }
+
+        /// <summary>The directory the project is in.</summary>
+        string Directory { get; }
+
+        /// <summary>
+        /// Multitargeting projects do "inner builds", one for each target framework. This property represents the
+        /// project without doing any of the per-target framework inner builds.
+        /// </summary>
+        public ITargetFramework OuterBuild { get; }
```

Comment:
Will this be null for VS?
Nominations only have inner builds.
Should that be called out?

Reply:
no, it will be the adapter's responsibility to "emulate" an outer build. We already have code that checks all the inner builds to make sure they have the same value, and throw if any inner build as a different value.  The CPS nomination adapter can use the same logic.

Reply:
I see. 
So, in VS, you'd have a "multi tfm" adapter that'd just get the values from every TFM and return correct value if it's the same and throw if they're different?

I'll see it when the PR comes :)